packages/hotspot/configure/src/hotspot-polygon.jsx

```diff
@@ -50,57 +66,100 @@ class PolComponent extends React.Component {
   };
 
   handleMouseEnter = () => {
+    this.setState({ hovered: true });
     document.body.style.cursor = 'pointer';
   };
 
   handleMouseLeave = () => {
+    this.setState({ hovered: false });
     document.body.style.cursor = 'default';
   };
 
-  handleOnDragEnd = (e) => {
+  handleOnDragEnd = (e, updateModel = false) => {
     const { onDragEnd, id } = this.props;
-    const points = e.target.points() || this.state.points;
+    const points = e.target.points() || this.serialize(this.state.points);
 
     const newPoints = points.reduce((acc, currentPointCoordinate, index) => {
       if (index % 2 === 0 && index + 1 < points.length) {
-        acc.push({
-          x: currentPointCoordinate + e.target.x(),
-          y: points[index + 1] + e.target.y(),
-        });
-
-        return acc;
-      } else {
-        return acc;
+        return [
+          ...acc,
+          {
+            x: currentPointCoordinate + e.target.x(),
+            y: points[index + 1] + e.target.y(),
+          },
+        ];
       }
+
+      return acc;
     }, []);
 
-    onDragEnd(id, {
-      points: newPoints,
-    });
+    this.setState({ points: newPoints, ...this.getOffset(newPoints) });
+
+    if (updateModel) {
+      onDragEnd(id, { points: newPoints });
+    }
+  };
+
+  handleOnDragVertex = (e, changedIndex, updateModel) => {
+    const { onDragEnd, id } = this.props;
+    const { points } = this.state;
+
+    const newPoints = points.map((point, index) =>
+      index === changedIndex ? { x: e.target.x(), y: e.target.y() } : point,
+    );
+
+    this.setState({ points: newPoints, ...this.getOffset(newPoints) });
+
+    if (updateModel) {
+      onDragEnd(id, { points: newPoints });
+    }
   };
 
   render() {
     const { classes, correct, hotspotColor, outlineColor, strokeWidth = 5 } = this.props;
-    const { points, x, y } = this.state;
+    const { points, x, y, hovered } = this.state;
+
+    const calculatedStrokeWidth = correct ? strokeWidth : hovered ? 1 : 0;
+    const calculatedStroke = correct ? outlineColor : hovered ? '#00BFFF' : '';
```

Comment:
Can you please store this color code somewhere? If we don't have it in theme, maybe at least declare a const for it?